bird_cloud_gnn/radar_dataset.py

```diff
@@ -57,10 +57,18 @@ def __init__(
         Raises:
             ValueError: If `data_folder` is not a valid folder.
         """
-        if not os.path.isdir(data_folder):
-            raise ValueError(f"'{data_folder}' is not a folder")
+
+        self.data_folder = None
+        self.input_data = None
+        if isinstance(data, pd.DataFrame):
+            self.input_data = data
+        else:
+            if os.path.isdir(data) | os.path.isfile(data):
+                self.data_folder = data
+            else:
+                raise ValueError("'data' is not a folder, file or pandas.DataFrame")
```

Comment:
I think here it would be better to have something like
```python
if os.path.isdir(data):
    self.data_folder = data
elif os.path.isfile(data):
    self.input_data = self._read_one_file(data)
else:
    # Error
```
So we handle the dir vs file case early on.

Reply:
Writing from the future, I see that my suggestion would affect the caching, essentially removing caching from the case where `data` is a file. I don't see that as an issue, because a single file read usually won't be too slow.

However, there an alternative here would be creating a new variable `input_file`, so we have three input-related variables (`input_data`, `input_file`, and `data_folder`. Two of them would always be None. This would also handle the dir vs file case early on while allowing for caching.

@lyashevska, do you have an opinion?

Reply:
This should be:
```python
if os.path.isdir(data):
    self.data_folder = data
elif os.path.isfile(data):
    self.input_data = pd.read_csv(data)
else:
    # Error
```
As `_read_one_file` directly processes it. However this would remove the parquet input option for single file

Reply:
Maybe renaming `data_folder` to `data_path` would already make things clearer

Reply:
Hi Bart, I understood better your idea as a whole. My main concern here was on separating early on whether the `data_folder` (or `data_path`) is a folder or file so that the `process` function is simpler.

So my only comment here is that maybe it makes sense to do one of the two:

- As you suggested, rename `data_folder` to `data_path`, because it can be either folder or file.
- Create `data_file` and `data_folder`. One will be None, and the other will point to the file or folder.

I'm more inclined to go with your suggestion of just renaming. I will continue the review with that assumption, let me know what you think.